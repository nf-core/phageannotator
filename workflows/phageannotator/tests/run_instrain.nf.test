nextflow_workflow {

    name "Test workflow: PHAGEANNOTATOR"
    script "workflows/phageannotator/main.nf"
    workflow "PHAGEANNOTATOR"

    // Dependencies
    tag "SEQKIT_SEQ"
    tag "QUALITYFILTERVIRUSES"
    tag "ANICLUSTER_ANICALC"
    tag "ANICLUSTER_ANICLUST"
    tag "ANICLUSTER_EXTRACTREPS"
    tag "COVERM_CONTIG"
    tag "INSTRAIN_STB"
    tag "FASTA_VIRUS_CLASSIFICATION_GENOMAD"
    tag "FASTA_VIRUS_QUALITY_CHECKV"
    tag "FASTA_ALL_V_ALL_BLAST"
    tag "FASTA_MICRODIVERSITY_INSTRAIN"
    tag "CAT_CAT"
    tag "BOWTIE2_BUILD"
    tag "GENOMAD_ENDTOEND"
    tag "GUNZIP"
    tag "FASTQ_ALIGN_BOWTIE2"


    test("Parameters: run_instrain = true") {
        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        [
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists:true),
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                        ]
                    ],
                    [
                        [ id:'test2' ],
                        [
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists:true),
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists:true)
                    ],
                    [
                        [ id:'test2' ],
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists:true)
                    ]
                )
                """
            }
            params {
                outdir                      = "$outputDir"
                run_instrain                = true
                // parameters to decrease sensitivity for test data
                mash_screen_min_score   = 0.01
                genomad_min_score       = 0.01
                genomad_max_fdr         = 1
                // speed up options since tools are fully tested in subworkflows
                genomad_disable_nn          = true
                genomad_sensitivity         = 0.1
                checkv_minimal_db           = true
                // lower instrain sensitivity so tool runs on test data
                instrain_min_ani            = 0.01
                instrain_min_mapq           = 2
                instrain_min_variant_cov    = 5
                instrain_min_snp_freq       = 0.01
                instrain_max_snp_fdr        = 1
                instrain_min_genome_cov     = 0.5
                instrain_popani_thresh      = 0.0
                instrain_min_genome_comp    = 0.001
                instrain_min_genome_breadth = 0.01
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert path("${outputDir}/VirusAbundance/coverm/contig/all_samples_alignment_results.tsv").readLines().size() == 4 }
            )
        }
    }
}
