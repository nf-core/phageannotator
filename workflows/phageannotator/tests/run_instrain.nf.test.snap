{
    "Parameters: run_instrain = true": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus_summary.tsv:md5,ae6d66b39f6ea43628db461e42012027"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test_virus_summary.tsv:md5,ae6d66b39f6ea43628db461e42012027"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test2"
                        },
                        "quality_summary.tsv:md5,5f1b0ace5264c11b2f5e8b0decbf2f9e"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "quality_summary.tsv:md5,5f1b0ace5264c11b2f5e8b0decbf2f9e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_cds_final_merged_output.tsv:md5,db5070088d1bea637525687595c03edb"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.IS_gene_info.tsv:md5,1d8ab275bc76669a149aa3802e67fc65"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.IS_gene_info.tsv:md5,1d8ab275bc76669a149aa3802e67fc65"
                    ]
                ],
                "9": [
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,04bf25e5231ce1f8077581bb7a2c07f4",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,1439e948f383899aaccbf49506923b04",
                    "versions.yml:md5,2d1864a0523e2738f150eaf87d4472d5",
                    "versions.yml:md5,374c63bf7c31c6c34acbf8d74e6c7dcc",
                    "versions.yml:md5,3c8b8ac5a86d6d7fba86fc61f4d5fca9",
                    "versions.yml:md5,5bf636608598d8b495d8b38fcee37dd0",
                    "versions.yml:md5,6323b255a9c1b9ed3e1a41aea5bfa2bb",
                    "versions.yml:md5,716410d7e9dc048a64ada71202cbc23e",
                    "versions.yml:md5,74b2068d24b17a2f4a097a3063da3dae",
                    "versions.yml:md5,769114e3bb009aceb4763a1a0fa80d5a",
                    "versions.yml:md5,7cb8c67c6ff9cf01f799d6730d5ebd15",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,8aebba726df29aad131772449019ee78",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,a856eefde99260259440d942c1d56ff4",
                    "versions.yml:md5,afc743a69ab04322fbc732b24e18d208",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072",
                    "versions.yml:md5,dc43c8404f9f95b46c1a9c8d51687032",
                    "versions.yml:md5,dc43c8404f9f95b46c1a9c8d51687032"
                ],
                "anicluster_reps_fna_gz": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "filtered_viruses_fna_gz": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "host_predictions_tsv": [
                    
                ],
                "instrain_gene_info": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.IS_gene_info.tsv:md5,1d8ab275bc76669a149aa3802e67fc65"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.IS_gene_info.tsv:md5,1d8ab275bc76669a149aa3802e67fc65"
                    ]
                ],
                "marker_taxonomy_tsv": [
                    
                ],
                "pharokka_output_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_cds_final_merged_output.tsv:md5,db5070088d1bea637525687595c03edb"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,04bf25e5231ce1f8077581bb7a2c07f4",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,1439e948f383899aaccbf49506923b04",
                    "versions.yml:md5,2d1864a0523e2738f150eaf87d4472d5",
                    "versions.yml:md5,374c63bf7c31c6c34acbf8d74e6c7dcc",
                    "versions.yml:md5,3c8b8ac5a86d6d7fba86fc61f4d5fca9",
                    "versions.yml:md5,5bf636608598d8b495d8b38fcee37dd0",
                    "versions.yml:md5,6323b255a9c1b9ed3e1a41aea5bfa2bb",
                    "versions.yml:md5,716410d7e9dc048a64ada71202cbc23e",
                    "versions.yml:md5,74b2068d24b17a2f4a097a3063da3dae",
                    "versions.yml:md5,769114e3bb009aceb4763a1a0fa80d5a",
                    "versions.yml:md5,7cb8c67c6ff9cf01f799d6730d5ebd15",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,8aebba726df29aad131772449019ee78",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,a856eefde99260259440d942c1d56ff4",
                    "versions.yml:md5,afc743a69ab04322fbc732b24e18d208",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072",
                    "versions.yml:md5,dc43c8404f9f95b46c1a9c8d51687032",
                    "versions.yml:md5,dc43c8404f9f95b46c1a9c8d51687032"
                ],
                "virus_classification_tsv": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2_virus_summary.tsv:md5,ae6d66b39f6ea43628db461e42012027"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test_virus_summary.tsv:md5,ae6d66b39f6ea43628db461e42012027"
                    ]
                ],
                "virus_enrichment_tsv": [
                    
                ],
                "virus_quality_tsv": [
                    [
                        {
                            "id": "test2"
                        },
                        "quality_summary.tsv:md5,5f1b0ace5264c11b2f5e8b0decbf2f9e"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "quality_summary.tsv:md5,5f1b0ace5264c11b2f5e8b0decbf2f9e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-16T22:28:04.553758819"
    }
}